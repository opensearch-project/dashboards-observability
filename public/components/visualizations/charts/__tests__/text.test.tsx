/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { waitFor } from '@testing-library/react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import { TEST_VISUALIZATIONS_DATA } from '../../../../../test/event_analytics_constants';
import { Text } from '../text/text';

describe('Text component', () => {
  configure({ adapter: new Adapter() });

  it('Renders text component', async () => {
    const wrapper = mount(<Text visualizations={TEST_VISUALIZATIONS_DATA} />);

    wrapper.update();

    await waitFor(() => {
      expect(
        toJson(wrapper, {
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });
});
