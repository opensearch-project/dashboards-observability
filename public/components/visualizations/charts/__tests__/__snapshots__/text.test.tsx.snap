// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Renders text component 1`] = `
<div
  data-test-subj="workspace__viz_markdown"
>
  <div
    className="euiMarkdownFormat"
  >
    <div>
      <h2
        key="h-2"
      >
        Text
      </h2>
      

      <p
        key="h-3"
      >
        Welcome to text editor!
      </p>
      

      <p
        key="h-4"
      >
        With text editor, you are able to add text description(s) to your dashboards.
      </p>
    </div>
  </div>
</div>
`;
